code/game/objects/structures/tables_racks.dm

```diff
@@ -35,6 +35,29 @@
 	var/deconstruction_ready = TRUE
 	var/flipped = 0
 
+/obj/structure/table/Bumped(mob/living/carbon/human/H)
+	. = ..()
+	if(!istype(H) || H.shoes || H.stat == DEAD || istype(H.dna.species, /datum/species/shadow/ling))
+		return ..()
+	if(prob(1))
+		to_chat(H, "<span class='warning'>You stub your toe on the [name]!</span>")
+		H.visible_message("[H] stubs their toe on the [name].")
```

Comment:
you can use visible message for both this and the message to the victim, check visible_message's second argument.